src/App.js

```diff
@@ -18,15 +18,99 @@ import {
   Route,
 } from "react-router-dom";
 
+function useOscillator(frequency, type) {
+  const oscRef = useRef(null);
+
+  useEffect(() => {
+    if (oscRef.current) {
+      oscRef.current.frequency.value = frequency;
+    }
+  }, [frequency]);
+
+  useEffect(() => {
+    if (oscRef.current) {
+      oscRef.current.type = type;
+    }
+  }, [type]);
+
+  return oscRef;
+}
```

Comment:
You should move all these custom hooks to separate files. Something like `src/hooks/useOscillator.js` for example